pkg/registry/namespace/etcd/etcd.go

```diff
@@ -119,9 +120,22 @@ func (r *REST) Delete(ctx api.Context, name string, options *api.DeleteOptions)
 		return nil, err
 	}
 
-	// upon first request to delete, we switch the phase to start namespace termination
+	// Decide whether we need to do an update to set deletion timestamp, termination phase, and (optionally) grace period seconds
+	needsUpdate := false
+	switch {
+	case namespace.DeletionTimestamp.IsZero():
+		needsUpdate = true
+	case namespace.Status.Phase != api.NamespaceTerminating:
+		needsUpdate = true
```

Comment:
@derekwaynecarr would like your opinion on this change. If the deletion time stamp is set, but the phase is not terminating, should we transition to terminating phase?